windows_docker_resources/Dockerfile

```diff
@@ -0,0 +1,58 @@
+# escape=`
+# This Dockerfile needs to be built from the parent directory (ros2/ci) so the build context
+# includes the python scripts
+
+# To find this value run in powershell:
+# $(Get-ItemProperty 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion').ReleaseId
+ARG WINDOWS_RELEASE_ID=1909
+
+# In order to ensure the image is correctly compatible with the host system a
+# more precise version than the release id is required. To find this value run
+# the following in powershell.
+# $(Get-ItemProperty -Path 'HKLM:SOFTWARE\Microsoft\Windows NT\CurrentVersion\Update\TargetingInfo\Installed\Server.OS.amd64' -Name Version).Version"
+# Fall back to the Release ID
+ARG WINDOWS_RELEASE_VERSION=$WINDOWS_RELEASE_ID
+
+# Indicates that the windows image will be used as the base image. Must be same or older than host.
+# --isolation=hyperv is needed for both build/run if the image id is older than the host id
+# Use --isolation=process if you need to build in a mounted volume
+FROM mcr.microsoft.com/windows:$WINDOWS_RELEASE_VERSION
+
+# Install cinc-solo, a compiled binary of chef-solo
+RUN powershell "iex  ((New-Object System.Net.WebClient).DownloadString('https://omnitruck.cinc.sh/install.ps1')); install -version 15.10.12"
+
+# Install Chocolatey by powershell script
+RUN powershell -noexit "Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))"
+
+# choco installs. chefdk is being installed to get berks and download cookbook dependencies
+RUN choco install -y git chefdk
+
+# Copy over necessary files into container
+RUN mkdir C:\TEMP
```

Comment:
This is a pretty commonly used directory, and I wouldn't be surprised if it randomly started showing up in docker images. Since `mkdir -p` doesn't work on Windows, it might be a good idea to use powershell to mimic `mkdir -p` behavior to possibly avoid breakage down the road.

Reply:
Updated the command to check if the directory already exists. I tested it out with a sample dockerfile in windows. Seems happy.